File: the_vowel_code.py
Dict = {}
Dict['vowels'] = ["a", "e", "i", "o", "u"]
Dict['numbers'] = ["1", "2", "3", "4", "5"]
print (Dict)

def encode(st):
    st_to_list = list(st)
    for i in range(len(st_to_list)):
        n = st_to_list[i]
        if n in Dict['vowels']:
            vowels_list=Dict['vowels']
            index = vowels_list.index(n)
            st_to_list[i] = Dict['numbers'][index]
    final_string = ''.join(st_to_list)
    return final_string
    
def decode(st):
    st_to_list = list(st)
    for i in range(len(st_to_list)):
        n = st_to_list[i]
        if n in Dict['numbers']:
            numbers_list=Dict['numbers']
            index = numbers_list.index(n)
            st_to_list[i] = Dict['vowels'][index]
    final_string = ''.join(st_to_list)
    return final_string

File: theVowelCode.js
/*
Step 1: Create a function called encode() to replace all the lowercase vowels in a given string with numbers according to the following pattern:

a -> 1
e -> 2
i -> 3
o -> 4
u -> 5
For example, encode("hello") would return "h2ll4". There is no need to worry about uppercase vowels in this kata.

Step 2: Now create a function called decode() to turn the numbers back into vowels according to the same pattern shown above.

For example, decode("h3 th2r2") would return "hi there".

For the sake of simplicity, you can assume that any numbers passed into the function will correspond to vowels.
*/

let vowels = ["a", "e", "i", "o", "u"];
let numbers = ["1", "2", "3", "4", "5"];

function encode(string) {
  var list = string.split('');
  for (let i=0; i<list.length; i++){
    if (vowels.includes(list[i])){
      list[i] = numbers[vowels.indexOf(list[i])];
    }
  }
  return list.join('');
}

function decode(string) {
  var list = string.split('');
  for (let i=0; i<list.length; i++){
    if (numbers.includes(list[i])){
      list[i] = vowels[numbers.indexOf(list[i])];
    }
  }
  return list.join('')
}